nsqd/protocol_v2.go

```diff
@@ -249,12 +252,24 @@ func (p *protocolV2) messagePump(client *clientV2, startedChan chan bool) {
 			// last iteration we flushed...
 			// do not select on the flusher ticker channel
 			memoryMsgChan = subChannel.memoryMsgChan
+			if subChannel.topologyZone == topologyZone && topologyZone != "" {
```

Comment:
Here's another little simplification idea: only one `localMsgChan`. If region+zone match, use zone chan, elif region match use region chan, else nil. The zone chan is always written before the region chan so it's redundant to listen on both.

Reply:
I like the way [Kubernetes frames this](https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/#topologykubernetesiozone)

> Kubernetes makes a few assumptions about the structure of zones and regions:
> 
> 1. regions and zones are hierarchical: zones are strict subsets of regions and no zone can be in 2 regions
> 1. zone names are unique across regions; for example region "africa-east-1" might be comprised of zones "africa-east-1a" and "africa-east-1b"

I'm thinking of zone labels in the same way - they are expected to be unique (which i think is how all cloud platforms treat this) Practically this means a client might specify only zone, and a nsqd might specify region & zone and that's ok. the zone matches or it doesn't. 

(this would be something to cover in docs of course)

Reply:
OK, that works fine. It still works with the suggestions to calculate "isLocalZone" and "isLocalRegion" boolean properties just once per identify, and have just one "localMsgChan" in `messagePump()` :)

Reply:
oh yes, definitely 👍 on some booleans to simplify this code